### ISSUE TITLE
Import additional organizational facets including for centers, divisions, and institutions

### ISSUE THREAD

**Issue by paulalbert1** [wcmc-its/vivodashboard#41]
VIVO Dashboard would be a lot more versatile if it captured additional affiliations

## Summary

| Org unit |  Workflow  | Update frequency |
| ----- | ----- | ---- |
| CTSC   | manually generated report from CTSC, load into DynamoDB  | semi-annual? |
| Other centers | ask administrators for role info, load into DynamoDB | semi-annual? |
| Divisions | query ED for start and stop dates, load into DynamoDB    | daily |
| Primary affiliation | query ED, assume start and stop dates correspond to academic roles, load into DynamoDB    | daily |
| Residency programs | manually generated report from CTSC, load into DynamoDB   | annual |
| MD-PhD alumni | obtain from spreadsheets    | one time?     |
| Grad school alumni | obtain from spreadsheets    | one time?    |

### CTSC

 to provide

### Cancer Center

The following CSV file contains a list of Cancer Center members with start dates and end dates.

[CancerCenterMembers-2018-11-29.txt](https://github.com/wcmc-its/vivodashboard/files/2630656/CancerCenterMembers-2018-11-29.txt)

Long-term, I'm hoping we can get this from ASMS directly and dynamically. But it would be important to get this out there sooner rather than later.

### Other centers

As per Dr. Cole's suggestion, it would be nice to have role data for these org units....

```
Caryl and Israel Englander Institute for Precision Medicine
https://ipm.weill.cornell.edu/
Contact: https://directory.weill.cornell.edu/person/profile/rok2011

Dalio Institute of Cardiovascular Imaging 
https://hearthealth.weillcornell.org/about-us/dalio-institute
Contact: https://directory.weill.cornell.edu/person/profile/ammatier ??

Cardiovascular Research Institute
https://cvri.weill.cornell.edu/
Contact: https://directory.weill.cornell.edu/person/profile/sjh2004

Gale and Ira Drukier Institute for Children's Health
https://drukierinstitute.weill.cornell.edu/
Contact: https://directory.weill.cornell.edu/person/profile/sjh2004

Ronald O. Perelman and Claudia Cohen Center for Reproductive Medicine
https://ivf.org/
Contact: https://directory.weill.cornell.edu/person/profile/tavazque

Joan and Sanford I. Weill Center for Metabolic Health
https://weill.cornell.edu/units/joan-and-sanford-i-weill-center-metabolic-health
Contact: https://directory.weill.cornell.edu/person/profile/sjh2004

Appel Institute
https://appel.weill.cornell.edu/
Contact: https://directory.weill.cornell.edu/person/profile/kbc2001
```

### Divisions

Prerequisite: ASMS assigns N codes. ED imports appointments from ASMS.

Requirements:
- Map division affiliations to start and end dates. Import those into wcmc_appointments and wcmc_departments.
- Import only select divisions. Ignore the less common ones.
- Output as custom labels, e.g., `Cardiology (Pediatrics)`.

The following IDs from wcmc_division would be considered in scope...
```
42
47
123
37
10000142
10000149
124
122
54
48
51
41
45
158
10000181
155
101
10000002
36
152
10000150
121
10000107
162
100
105
10000102
10000105
10000006
92
97
10000103
94
61
10000124
10000221
163
68
106
98
135
71
91
9
10000141
10000161
150
102
140
99
```

### Primary affiliation

Curt asked that we include WCM-Q as an affiliation. We don't have historical affiliation data in ED, so this isn't ideal. 

We have a few options:
1. Create a new facet independent of the other ones: `Primary affiliation`. This would only be for the current point in time
2. Query ASMS to determine users' earliest start and latest end date, and then pair it with their primary affiliation. Add this to the Organization facet. This is not ideal.
3. Infer start and end date for institutional affiliation but only for currently active users. Add these start and end dates to wcmc_appointments. Add this role to the Organization facet. 

### Residency programs

Can get start and end date from GME.

Syntax: `Medicine (Residency Program)`

### MD-PhD alumni 

Paul has a spreadsheet for existing alumni. Many do not have CWIDs we would have to create some.

### Grad school alumni 

Would have to get data.

**Comment by paulalbert1** [comment]
This is done.
